### ISSUE TITLE
Heat gains of whole GFA and whole envelope are allocated to conditioned space

### ISSUE THREAD

**Issue by gabriel-happle** [architecture-building-systems/CityEnergyAnalyst#1633]
Some smaller buildings in the new TPW case study are overheating and resulting in very unrealistic conditions.

I think the reason is that their air-conditioned area is relatively small compared to the GFA (small `Hs` value).

If all buildings are defined as mixed use with a high residential fraction in Singapore, typically `0.25 < Hs < 0.65`.

This means that all solar heat gains and all appliance and person heat and humidity gains are concentrated on 25% of the space, leading to very high cooling loads and in extreme cases the code of the RC-model to break down.

I think a solution would be to only account for a fraction of the heat gains in such buildings.
However, I am not sure yet. I'm still investigating the situation.
As I remember there was already a similar issue raised by @martin-mosteiro earlier.

@JIMENOFONSECA, @BhargavaKrishnaSreepathi, @shizhongming, @shanshanhsieh I think we should address this before we run simulations for our upcoming publications. What could be the action plan?

**Comment by JIMENOFONSECA** [comment]
Yes I think we made a previous work around that, and it somehow broke Martin’s case study.

I imagine creating again a new factor for the amount of electrified area (responsible of internal loads). this should be different to Hs.


On 2 Oct 2018, at 5:12 PM, Gabriel Happle <notifications@github.com<mailto:notifications@github.com>> wrote:


Some smaller buildings in the new TPW case study are overheating and resulting in very unrealistic conditions.

I think the reason is that their air-conditioned area is relatively small compared to the GFA (small Hs value).

If all buildings are defined as mixed use with a high residential fraction in Singapore, typically 0.25 < Hs < 0.65.

This means that all solar heat gains and all appliance and person heat and humidity gains are concentrated on 25% of the space, leading to very high cooling loads and in extreme cases the code of the RC-model to break down.

I think a solution would be to only account for a fraction of the heat gains in such buildings.
However, I am not sure yet. I'm still investigating the situation.
As I remember there was already a similar issue raised by @martin-mosteiro<https://github.com/martin-mosteiro> earlier.

@JIMENOFONSECA<https://github.com/JIMENOFONSECA>, @BhargavaKrishnaSreepathi<https://github.com/BhargavaKrishnaSreepathi>, @shizhongming<https://github.com/shizhongming>, @shanshanhsieh<https://github.com/shanshanhsieh> I think we should address this before we run simulations for our upcoming publications. What could be the action plan?

**Comment by martin-mosteiro** [comment]
@gabriel-happle Yep, I had the same issue. The reason was (1) the electrified area is hard-coded as 90% of the total area, and (2) all the electrical gains get dumped in the conditioned area, regardless of whether the electrified area and conditioned areas match.

As I mentioned before, I think we need to choose a consistent unit of analysis and base our calculations on that - right now we use conditioned, electrified and gross floor area for different calculations in a confusing (and also inconsistent) way.

**Comment by JIMENOFONSECA** [comment]
Oh. So that means it has to do with the assumed geometry of the building and the real envelope that is heated. Mmm. I wonder if Hs should then be applied proportionately to walls, windows and even roof areas?

Of course it should be coherent with the solar radiation that the area gets i guess

____________


Dr. Jimeno A. Fonseca
Senior Researcher and Project Coordinator
Multi-Scale Energy Systems for Low-Carbon Cities (MuSES)
Future Cities Laboratory
Singapore ETH-Centre

Chair of Architecture and Building Systems
Institute of Technology in Architecture
ETH Zurich

Scientific Advisory Committee
Energy and Sustainability 2017
On 2 Oct 2018, 18:25 +0800, Gabriel Happle <notifications@github.com>, wrote:

@JIMENOFONSECA<https://github.com/JIMENOFONSECA> and @martin-mosteiro<https://github.com/martin-mosteiro>, after closer investigation, I think the issue is more related to the geometry calculations of the RC-model.
I.e., the heat exchange surfaces are really high compared to the conditioned area, basically they are impossible geometries.

  *   Af = 113m2

  *   Awin = 150m2

  *   Awall = 360m2

E.g., when I correct the thermal transmittance coefficients for the building that I am testing with a value in the range of Hs, it seems to produce reasonable results.

What I did is in building_properties.calc_prop_rc_model() to lower the values of df['Htr_w'] and df['HD'] for testing purposes.

**Comment by gabriel-happle** [comment]
It seems the RC-model gets out of bounds for certain small buildings. In the case study that @shizhongming and @BhargavaKrishnaSreepathi are testing, this only happened to one single building (the smallest one) out of more than 500.

I can create a warning for now that stops the calculation and gives a specific error message when the RC-model goes crazy.

As for the geometry calculations, I think this requires some bigger effort. Maybe we can decide in the next developers meeting how to address it.

**Comment by daren-thomas** [comment]
@martin-mosteiro is this fixed? if so, please close for M2.9.2

**Comment by gabriel-happle** [comment]
re-opening this!

**Comment by JIMENOFONSECA** [comment]
After our discussion with @gabriel-happle and @lguilhermers we are going to try to change the internal loads, solar radiation loads, and heat transfer coefficients to account for the area of air-conditioned space. 

@gabriel-happle will do the honors and @lguilhermers and myself will test

**Comment by gabriel-happle** [comment]
@JIMENOFONSECA and @lguilhermers,
also @shanshanhsieh and @martin-mosteiro and @shizhongming,
Please check out branch https://github.com/architecture-building-systems/CityEnergyAnalyst/tree/i1633-gains-and-losses-proportional-to-conditioned-area

I made two changes:
- Heat transfer areas of the RC model are weighted by the fraction of conditioned area `Hs`. Roof and floor/basement are weighted directly with `Hs`, windows and walls with `sqrt(Hs)`

- In the RC model calculation itself, internal gains from electric lights and appliances are weighted by `Hs/Es` and solar gains are weighted with `sqrt(Hs)`. This second part avoids changing any solar calculations in earlier steps.

In the test case, the following changes:
`B01`
![image](https://user-images.githubusercontent.com/17002288/66126994-540ba480-e61d-11e9-9657-36039cb576e7.png)

`all buildings`
![image](https://user-images.githubusercontent.com/17002288/66127057-73a2cd00-e61d-11e9-98c2-4abcea3fdcce.png)

changes remain small in buildings with high `Hs`.
The buildings that are most affected are

**Comment by gabriel-happle** [comment]
just a thought: maybe we should weigh the heat exchange area with `Hs/Ns` so that it remains `=1` for most buildings. 
I'm doing that. It makes a bit more sense since buildings with `Hs=0.9` probably exchange over the whole envelope.

![image](https://user-images.githubusercontent.com/17002288/66128703-ff6a2880-e620-11e9-879b-b3301a27200e.png)
![image](https://user-images.githubusercontent.com/17002288/66128767-2163ab00-e621-11e9-9154-c2d3eaf6de72.png)

**Comment by martin-mosteiro** [comment]
@gabriel-happle Thanks for the first stab at this. Some points:

1. `NFA = GFA * Ns`. For the Swiss case, it's safe to assume the Net Floor Area is equal to the conditioned area. But in Singapore there are unconditioned, actively occupied floor areas (Hs < Ns). We could probably use `Es` as a proxy though: assume only the net floor area is electrified, and thus `NFA = GFA * Es`, with the conditioned area being `Af = GFA * Hs`.
2. 
```
El = tsd['El'][t] * (bpr.architecture.Hs / bpr.architecture.Es)  # account for a proportion of internal gains
Ea = tsd['Ea'][t] * (bpr.architecture.Hs / bpr.architecture.Es)  # account for a proportion of internal gains
```
Since `tsd['El'][t] = schedule['El_Wm2'] * bpr.rc_model.Aef_m2 = schedule['El_Wm2'] * bpr.rc_model.GFA_m2 * bpr.architecture.Es` your proposed change is exactly the same as doing `El = schedule['El_Wm2'] * bpr.rc_model.Af_m2`. Which makes sense, gains should only be for the electricity actually used in the room. Minor point: right now in CEA it's possible to specify a building with `Es < Hs` (i.e., the conditioned floor area is larger than the electrified area). This instinctively doesn't seem realistic, but in the current implementation it would lead to the internal gains being larger than the actual electricity consumed in the building.

3.
```
I_sol = tsd['I_sol_and_I_rad'][t] * np.sqrt(bpr.architecture.Hs)  # account for a proportion of solar gains
```
This is a pretty big simplification... I'm not sure how this would affect buildings in the Swiss case, where the share of GFA (and hence heated area) to unoccupied area is quite significant (~50% for lots of buildings).

**Comment by martin-mosteiro** [comment]
Especially since GFA includes underground floors. Imagine the following (hypothetical) case:
One building with 3 floors above ground and 1 unheated floor below ground. GFA = 4 * footprint. Hs = 0.75. However the solar gains should indeed all be assigned to the top 3 floors! So your "fix" would underestimate solar gains by 25%.

**Comment by JIMENOFONSECA** [comment]
wait...should not Af be calculated as (GFA [above ground] * hs)? I believe that should be the implementation right?

**Comment by martin-mosteiro** [comment]
It isn't:
building_properties.py (line 382): `df['floors'] = df['floors_bg'] + df['floors_ag']`
building_properties.py (line 385): `df['GFA_m2'] = df['footprint'] * df['floors']  # gross floor area`
building_properties.py (line 274): `df['Af'] = df['GFA_m2'] * df['Hs']  # conditioned area: areas that are heated/cooled`

And it shouldn't, right? There can also be conditioned floor areas below ground, both in SG (say, in a shopping mall's underground shopping areas) and in CH (for example in the Hochschulquartier, the gym in the Polyterrasse).

**Comment by JIMENOFONSECA** [comment]
maybe we wait for the schedule maker and see how to implement (improve)? what is your take on this @gabriel-happle

**Comment by gabriel-happle** [comment]
This is not so straightforward as we thought.
Plus, the residential buildings in Singapore are an exception, with their very small `Hs`.
Maybe for the moment, we shouldn't change anything until we come up with a solution on how to account for the solar heat gains.

Judging from my own experience living in Singapore, the inside of a well-ventilated residential building is pretty much the outside temperature... unless you live in a condominium with ridiculously large windows.
From that logic, the solar heat gains in a residential building should only be accounted for when the windows are closed and the air-conditioning is on. Then it's probably easier to model air-conditioning like an appliance.

Another way would be to consider a fraction of the solar heat gains according to a threshold of `Hs`.
E.g., when `Hs` < 0.6 then we consider 75% of solar heat gains.... etc. But I haven't looked into this.

**Comment by gabriel-happle** [comment]
The "multiplier" of solar heat gains could also be an advanced user input that is set to 1` by default.

**Comment by JIMENOFONSECA** [comment]
@martin-mosteiro so we were discussing with @gabriel-happle . Maybe if we implement Hs_ag and Hs_bg we can take into account this variation of air-conditioned spaces above ground and underground.

basically having a proportion of area air-conditioned above ground and below ground. The default value will be 00.0 for Hs_bg.

What do you think:

```
El = tsd['El'][t] * (bpr.architecture.Hs_ag*GFA_ag + bpr.architecture.Hs_ag*GFA_bg  / bpr.architecture.Es)  # account for a proportion of internal gains
Ea = tsd['Ea'][t] * (bpr.architecture.Hs_ag*GFA_ag + bpr.architecture.Hs_ag*GFA_bg  / bpr.architecture.Es)  # account for a proportion of internal gains
I_sol = tsd['I_sol_and_I_rad'][t] * np.sqrt(bpr.architecture.Hs_ag)  # account for a proportion of solar gains
```

and do something similar with the values of df['Htr_w'] and df['HD'] for testing purposes.

**Comment by martin-mosteiro** [comment]
@gabriel-happle This appears to have been addressed by #2420. Please reopen if necessary.
